src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/NoOpRestoreUtilities.cs

```diff
@@ -158,6 +161,31 @@ internal static bool VerifyRestoreOutput(RestoreRequest request, CacheFile cache
                 return false;
             }
 
+            if (request.ForceUpdatePackageLastAccessTime)
+            {
+                foreach (var package in cacheFile.ExpectedPackageFilePaths)
+                {
+                    var packageFolder = new DirectoryInfo(Directory.GetParent(package).FullName).Parent.Parent.FullName;
+
+                    // we avoid fallback folders as they are readonly
+                    if (request.Project.RestoreMetadata.FallbackFolders.Any(fallbackFolder => fallbackFolder.Equals(packageFolder, StringComparison.OrdinalIgnoreCase)))
```

Comment:
path comparision needs to be environment specific. 

See: https://github.com/NuGet/NuGet.Client/blob/45bc94385cb980e463736fe753977b23efe92009/src/NuGet.Core/NuGet.Common/PathUtil/PathUtility.cs#L25